### PR TITLE
Get rid of trivial anonymous functions (part 3)

### PR DESCRIPTION
**Pull request by lgoettgens**
See https://github.com/oscar-system/Oscar.jl/pull/4126 for part 1 and the reasoning behind these changes.

### CODE REVIEW

File: src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/betti_numbers.jl [hunk 1]
```diff
@@ -34,7 +34,7 @@ function betti_number(v::NormalToricVarietyType, i::Int)
     end
     
     # extract vector of currently-known Betti numbers (or create it if necessary)
-    betti_numbers = get_attribute!(() -> fill(ZZRingElem(-1), d+1), v, :betti_number)::Vector{ZZRingElem}
+    betti_numbers = get_attribute!(v, :betti_number, [ZZ(-1) for _ in 1:(d+1)])::Vector{ZZRingElem}
```
Comment (fingolfin) [inline src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/betti_numbers.jl]: This always allocates `[ZZ(-1) for _ in 1:(d+1)]` and if the attribute is already set, discards it.

File: src/AlgebraicGeometry/ToricVarieties/ToricMorphisms/attributes.jl [hunk 2]
```diff
@@ -76,7 +76,7 @@ Map
     mapping_matrix = matrix(ZZ, zeros(ZZ, torsion_free_rank(torusinvariant_weil_divisor_group(cod)), 0))
     for i in 1:nrows(images)
       v = [images[i,k] for k in 1:ncols(images)]
-      j = findfirst(x -> x == true, [(v in maximal_cones(cod)[j]) for j in 1:n_maximal_cones(cod)])
+      j = findfirst(j -> v in maximal_cones(cod)[j], 1:n_maximal_cones(cod))
```
Comment (fingolfin) [inline src/AlgebraicGeometry/ToricVarieties/ToricMorphisms/attributes.jl:79]: nice

File: src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/standard_constructions.jl [hunk 3]
```diff
@@ -303,8 +302,7 @@ Ideal generated by
 function normal_toric_varieties_from_star_triangulations(P::Polyhedron)
   # find position of origin in the lattices points of the polyhedron P
   pts = matrix(ZZ, lattice_points(P))
-  zero = [0 for i in 1:ambient_dim(P)]
-  indices = findall(k -> pts[k:k,:] == matrix(ZZ, [zero]), 1:nrows(pts))
+  indices = findall(k -> is_zero_row(pts, k), 1:nrows(pts))
```
Comment (fingolfin) [inline src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/standard_constructions.jl:305]: nice!

**Review by fingolfin — APPROVED** [review]

File: src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/betti_numbers.jl (hunk 1 above)
Comment (lgoettgens) [reply src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/betti_numbers.jl]:
```suggestion
    betti_numbers = get_attribute!(() -> [ZZ(-1) for _ in 1:(d+1)], v, :betti_number)::Vector{ZZRingElem}
```
True. So let's revert this (up to the aliasing issue)

**Pull request merged**